flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamTaskITCase.java

```diff
@@ -110,6 +116,42 @@ public void testFailInEndOfConstructor() throws Exception {
                                 .build(EXECUTOR_RESOURCE.getExecutor()));
     }
 
+    @Test
+    public void testAsyncExceptionCanBeHandledDuringRestoring() throws Exception {
```

Comment:
I don't think that it is the right place for this test. Your test is just a simple unit test while this class is for IT tests. So I think `StreamTaskTest` is a better place for this test.

Reply:
@akalash Thanks for the comments. I fine to it & update it.